web/src/providers/SetupConfig/SetupConfig.provider.tsx

```diff
@@ -0,0 +1,84 @@
+import {noop} from 'lodash';
+import {createContext, useCallback, useContext, useMemo} from 'react';
+import {useUpdateConfigMutation} from 'redux/apis/TraceTest.api';
+import {useAppDispatch, useAppSelector} from 'redux/hooks';
+import {initialState, reset, setDraftConfig, setIsFormValid} from 'redux/setupConfig/slice';
+import SetupConfigSelectors from 'redux/setupConfig/selectors';
+import {ISetupConfigState, TDraftConfig} from 'types/Config.types';
+import SetupConfigService from 'services/SetupConfig.service';
+
+interface IContext extends ISetupConfigState {
+  isLoading: boolean;
+  onSaveConfig(draft: TDraftConfig): void;
+  onUpdateDraft(draft: TDraftConfig): void;
+  onReset(): void;
+  onIsFormValid(isValid: boolean): void;
+}
+
+export const Context = createContext<IContext>({
+  ...initialState,
+  isLoading: false,
+  onSaveConfig: noop,
+  onUpdateDraft: noop,
+  onIsFormValid: noop,
+  onReset: noop,
+});
+
+interface IProps {
+  children: React.ReactNode;
+}
+
+export const useSetupConfig = () => useContext(Context);
+
+const SetupConfigProvider = ({children}: IProps) => {
```

Comment:
What is the reason for having a new Provider? Could this be part of the `Config.provider`?

Reply:
I was thinking that the ConfigProvider would be for read-only, while the setup config one is the one we'll be using to manipulate the config draft for each of the tabs and sections of the settings page. Does it make sense?